### DISCUSSION TITLE
Wazuh in a Distributed installation

### DISCUSSION THREAD

**Discussion by mskarshinski** [Security-Onion-Solutions/securityonion#7676] [Unsupported Versions]
After almost a year of running SO, I am finally switching over to a distributed setup.  I was pretty severely overloading my standalone once I started adding more than about 10 Wazuh/ossec clients.  Lots of logs coming in.  I need to scale this installation up so that I can monitor 100 Wazuh agents on user workstations and servers.

I have setup one manager, one search node, and two sensors (forwarders).  My plan is to use one sensor to monitor Suricata (got this to work easily) and the other sensor to accept traffic from my workstation/server Wazuh agents.  I'm not sure if it works this way, can I point my wazuh agents at a sensor/forwarder, or only at the manager?  I became suspicious that I didn't fully understand this process when I tried to run so-allow on the sensor, firewall rules don't appear to exist there.  Am I allowing traffic for all minions when I run so-allow on the Manager?

Thank you to the entire SO team for a great, usable, flexible, and well-documented product, this thing is a MONSTER.

**Comment by cm-ops** [comment]
All Security Onion nodes that run Wazuh are running the Wazuh components: manager, API, and agent. 

To point the agents to an **SO Sensor node**, you can add the agent on the **SO Sensor** with `so-wazuh-agent-manage` Then to set the firewall settings on the sensor you would do the following on your **SO Manager**:
1. Create the firewall entry for the agent with `sudo so-firewall includehost wazuh_agent <IP>`
2. Edit the minion pillar file for the sensor you want to send the logs to `/opt/so/saltstack/local/pillar/minions/*sensor.sls` This will apply to only the minion you have the settings configured for.
3. Add the following to the pillar file
```
		firewall:
		  assigned_hostgroups:
		    chain:
		      DOCKER-USER:
		        hostgroups:
		          wazuh_agent:
		            portgroups:
		              - portgroups.wazuh_agent
```
4. Apply the firewall state to the sensor with `sudo salt '*sensor*' state.apply firewall`
5. Start/Restart the agent on the endpoint and check the logs for connection

**Comment by mskarshinski** [comment]
Thank you SO much for this, incredibly helpful.  I will take a swing at this over next weekend!

**Reply by mskarshinski** [reply]
Firewall changes did the trick!  After pushing a new ossec.conf to my agents, they are now checking in with my dedicated Wazuh sensor Node, and they are showing in the SOC.  New agents however, cannot register, I suspect 1515 is still blocked.  Is this the proper /opt/so/saltstack/local/pillar/minions/{my Wazuh sensor}_sensor.sls? (I added the wazuh_authd section after the wazuh_agent you suggested):

firewall:
  assigned_hostgroups:
    chain:
      DOCKER-USER:
        hostgroups:
          wazuh_agent:
            portgroups:
             - portgroups.wazuh_agent
          **wazuh_authd:
            portgroups:
             - portgroups.wazuh_authd**

Wazuh was previously on the Manager Node, in order to use the new dedicated Node, I copied over the /nsm/wazuh/etc files:
  - authd.pass (I am registering agents with password auth for now)
  - client.keys - the full list of previously registered ossec agents, substituting the local host's 001 key for the one in the file copied from the manager (the Manager's key).

Am I correct that each computer running Wazuh should be handling its own local ossec agent, then passing that info to the Manager?  Or, should I kill Wazuh on all boxes except for my "Wazuh Node" and register their ossec agents with my dedicated Wazuh manager?

Fleet dedicated nodes don't appear to install Wazuh, can I just register ossec on the Fleet node with my Wazuh node?

Again, I appreciate the help and am willing to research, if there is additional technical documentation that I'm missing, feel free to point me in that direction.

**Comment by mskarshinski** [comment]
Agent registration issue fixed, I forgot to add the following to /var/ossec/etc/ossec.conf:

<auth>
  <use_password>yes</use_password>
</auth>

I did make the firewall changes in my last post above, but I'll need to pull my "wazuh_agentd" section to see if it is actually required.
